content-api/content-actors/src/main/scala/org/sunbird/content/dial/DIALManager.scala

```diff
@@ -59,14 +59,20 @@ object DIALManager {
 
 	def validateAndGetRequestMap(channelId: String, requestList: List[Map[String, List[String]]])(implicit oec:OntologyEngineContext): Map[String, List[String]] = {
 		var reqMap = HashMap[String, List[String]]()
-		requestList.foreach(req => {
-			val contents: List[String] = req(DIALConstants.IDENTIFIER)
-			val dialcodes: List[String] = req(DIALConstants.DIALCODE)
-			validateReqStructure(dialcodes, contents)
-			contents.foreach(id => reqMap += (id -> dialcodes))
-		})
+		try {
+			requestList.foreach(req => {
+				val contents: List[String] = req(DIALConstants.IDENTIFIER)
+				val dialcodes: List[String] = req(DIALConstants.DIALCODE)
+				validateReqStructure(dialcodes, contents)
+				contents.foreach(id => reqMap += (id -> dialcodes))
+			})
+		} catch {
+			case e: Exception => throw new ClientException(DIALErrors.ERR_DIALCODE_CONTENT_LINK_FIELDS_MISSING, DIALErrors.ERR_DIALCODE_CONTENT_LINK_FIELDS_MISSING_MSG)
+		}
+		TelemetryManager.info("DIALManager::validateAndGetRequestMap:: requestList: " + requestList)
```

Comment:
Log any api failures/exceptions as TelemetryManager.warn message for ease of discovery

Reply:
ClientException gets logged. This is enough for finding out the issue.

Reply:
Added Exceptions log themselves. No need to log separately. I will remove info logs and recommit once QA testing is completed. They have been added for debugging purposes only.